src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeCache.targets

```diff
@@ -23,11 +23,127 @@ Copyright (c) .NET Foundation. All rights reserved.
   </PropertyGroup>
 
   <Target Name="ComposeCache"
-          DependsOnTargets="PrepareForComposeCache;
-                            RestoreForComposeCache;
-                            ComputeAndCopyFilesToCacheDirectory"/>
+          DependsOnTargets="_InitializeBasicProps;
+                            PrepareForComposeCache;
+                            PrepOptimizer;
+                            CacheWorkerMain;
+                            CacheFinalizer;"/>
+
+
+  <!--
+    ============================================================
+                                        CacheWorkerMain
+
+   Spins of the compose mappers
+    ============================================================
+    -->
+
+  <Target Name="CacheWorkerMain">
+    
+    <ItemGroup>
+      <_AllProjects Include="$(AdditionalProjects.Split('%3B'))"/>
+      <_AllProjects Include ="$(MSBuildProjectFullPath)"/>
+    </ItemGroup>
+    
+    <MSBuild Projects="%(_AllProjects.Identity)"
+                 Targets="CacheWorkerMapper"
+                 Properties="ComposeWorkingDir=$(ComposeWorkingDir);
+                             PublishDir=$(PublishDir);
+                             TargetFramework=$(_TFM);
+                             JitPath=$(JitPath);
+                             Crossgen=$(Crossgen);
+                             DisableImplicitFrameworkReferences=true;
+                             SkipUnchangedFiles=$(SkipUnchangedFiles);
+                             PreserveCacheLayout=$(PreserveCacheLayout)"
+                  ContinueOnError="WarnAndContinue"  />
+   </Target>
+  <!--
+    ============================================================
+                                        CacheWorkerMapper
+
+   Spins of the compose mappers
```

Comment:
This is the exact same description as CacheWorkerMain.  Since they are separate targets, they should have separate descriptions.